csharp/EmsApi.Client/V2/Service/EmsApiService.cs

```diff
@@ -0,0 +1,288 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using Refit;
+
+using EmsApi.Client.V2.Wrappers;
+
+
+namespace EmsApi.Client.V2
+{
+	using AuthCallbackDictionary = Dictionary<Action<string>, EventHandler<AuthenticationFailedEventArgs>>;
+	using ExceptionCallbackDictionary = Dictionary<Action<string>, EventHandler<ApiExceptionEventArgs>>;
+
+	/// <summary>
+	/// The client side representation of the EMS API.
+	/// </summary>
+	/// <remarks>
+	/// This assembly targets .NET standard, which is supported on all
+	/// flavors of .NET (Framework, Core, Xamarin, etc).
+	/// </remarks>
+	public class EmsApiService : IDisposable
+    {
+        /// <summary>
+        /// Provides access to the EMS API. The <seealso cref="ServiceConfig"/> property
+        /// must be set on this class before any API methods can be used.
+        /// </summary>
+        public EmsApiService()
+        {
+            m_config = new EmsApiServiceConfiguration();
+            Initialize();
+        }
+
+        /// <summary>
+        /// Provides access to the EMS API using the provided configuration settings.
+        /// At a minimum, a username and password must be specified.
+        /// </summary>
+		public EmsApiService( EmsApiServiceConfiguration config )
+        {
+			ValidateConfigOrThrow( config );
+			m_config = config;
+			Initialize();
+		}
+
+		/// <summary>
+		/// Sets up our API interface and wrapper classes.
+		/// </summary>
+		private void Initialize()
+		{
+			// Set up our event maps.
+			m_authCallbacks = new AuthCallbackDictionary();
+			m_exceptionCallbacks = new ExceptionCallbackDictionary();
+
+			// Set up authentication.
+			m_authHandler = new Authentication.EmsApiTokenHandler( m_config );
+			m_apiClient = new HttpClient( m_authHandler );
+			m_apiClient.BaseAddress = new Uri( m_config.Endpoint );
+
+			// Set up the API client abstraction.
+			m_api = RestService.For<IEmsApi>( m_apiClient );
+
+			// Set up wrapper properties for the interface.
+			EmsSystems = new EmsSystemWrapper( m_api );
+
+			// Subscribe to failure events.
+			EmsApiRouteWrapper.ApiMethodFailedEvent += ApiExceptionHandler;
+			m_authHandler.AuthenticationFailedEvent += AuthenticationFailedHandler;
+		}
+
+		/// <summary>
+		/// Gets or sets the current service configuration. If the endpoint or authorization
+		/// properties change, <seealso cref="RequestAuthentication"/> should be called.
+		/// </summary>
+		public EmsApiServiceConfiguration ServiceConfig
+        {
+            get { return m_config; }
+            set
+            {
+				ValidateConfigOrThrow( value );
+				m_config = value;
+				m_apiClient.BaseAddress = new Uri( m_config.Endpoint );
+				m_authHandler.UpdateConfiguration( m_config );
+            }
+        }
```

Comment:
Funky spacing here, too. Reading through the rest of the file, it looks like your auto-formatting might be not firing? or maybe you are crazy like Meg and don't use auto-formatting?